api/src/test/java/org/openmrs/liquibase/ChangeLogDetectiveTest.java

```diff
@@ -0,0 +1,134 @@
+import java.util.Map;
+import liquibase.changelog.ChangeSet;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class ChangeLogDetectiveTest {
+	@Test
+	public void shouldGetSnapshotVersionsInDescendingOrder() {
+		Map<String, List<String>> snapshotCombinations = new HashMap<>();
+		snapshotCombinations.put("1.9.x", null);
+		snapshotCombinations.put("2.4.x", null);
+		snapshotCombinations.put("2.1.x", null);
+		snapshotCombinations.put("2.2.x", null);
+		snapshotCombinations.put("2.3.x", null);
+		
+		ChangeLogDetective changeLogDetective = new ChangeLogDetective();
+		List<String> actual = changeLogDetective.getSnapshotVersionsInDescendingOrder(snapshotCombinations);
+		List<String> expected = Arrays.asList("2.4.x", "2.3.x", "2.2.x", "2.1.x", "1.9.x");
+		
+		assertEquals(expected, actual);
+	}
+	
+	@Test
+	public void shouldIgnoreDisableForeignKeyChecks() {
+		ChangeSet changeSet = mock(ChangeSet.class);
+		when(changeSet.getAuthor()).thenReturn("ben");
+		when(changeSet.getId()).thenReturn("disable-foreign-key-checks");
+		
+		ChangeLogDetective changeLogDetective = new ChangeLogDetective();
+		boolean actual = changeLogDetective
+		        .isVintageChangeSet("org/openmrs/liquibase/snapshots/core-data/liquibase-core-data-1.9.x.xml", changeSet);
+		assertTrue(actual);
+	}
+
+	@Test
+	public void shouldIgnoreEnableForeignKeyChecks() {
+		ChangeSet changeSet = mock(ChangeSet.class);
+		when(changeSet.getAuthor()).thenReturn("ben");
+		when(changeSet.getId()).thenReturn("enable-foreign-key-checks");
+		
+		ChangeLogDetective changeLogDetective = new ChangeLogDetective();
+		boolean actual = changeLogDetective
+		        .isVintageChangeSet("org/openmrs/liquibase/snapshots/core-data/liquibase-core-data-1.9.x.xml", changeSet);
+		assertTrue(actual);
+	}
+
+	@Test
+	public void shouldNotIgnoreDisableForeignKeyChecksForFilenameOtherThanLiquibaseCoreData_1_9_x() {
+		ChangeSet changeSet = mock(ChangeSet.class);
+		when(changeSet.getAuthor()).thenReturn("ben");
+		when(changeSet.getId()).thenReturn("disable-foreign-key-checks");
+
+		ChangeLogDetective changeLogDetective = new ChangeLogDetective();
+		boolean actual = changeLogDetective
+			.isVintageChangeSet("any_filename", changeSet);
+		assertFalse(actual);
+	}
+
+	@Test
+	public void shouldNotIgnoreEnableForeignKeyChecksForFilenameOtherThanLiquibaseCoreData_1_9_x() {
+		ChangeSet changeSet = mock(ChangeSet.class);
+		when(changeSet.getAuthor()).thenReturn("ben");
+		when(changeSet.getId()).thenReturn("enable-foreign-key-checks");
+
+		ChangeLogDetective changeLogDetective = new ChangeLogDetective();
+		boolean actual = changeLogDetective
+			.isVintageChangeSet("any_filename", changeSet);
+		assertFalse(actual);
+	}
+
+	@Test
+	public void shouldNotIgnoreOtherChangeSetFromBen() {
```

Comment:
Whatsup with changesets from Ben. Why do they receive special treatment? Am asking this from the perspective of some one looking at this code for the first time?

Reply:
I am adding a note to this test case.

Reply:
Done.